### PR TITLE
Bug 2076614: Expose ResourceEventStream Component as part of the Core SDK

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot of the demo plugin:
Screenshot from 2022-04-04 18-42-02

### CODE REVIEW

**Comment by spadgett** [comment]
@bipuladh What is the use case?

If it's for a resource page tab, I think we need a way to add standard tabs to your pages (YAML, environment vars, events, etc.). Then we wouldn't need to expose every component individually, and it would be less work for the plugins to include these.

cc @christianvogt

**Comment by bipuladh** [comment]
> @bipuladh What is the use case?
> 
> If it's for a resource page tab, I think we need a way to add standard tabs to your pages (YAML, environment vars, events, etc.). Then we wouldn't need to expose every component individually, and it would be less work for the plugins to include these.
> 
> cc @christianvogt

@spadgett the use case is for the details page. For most ResourcePages I guess YAML and Events are enough. I could look  into exposing the DetailsPage component and we can hold this PR. However, exposing these two components should allow most resource pages to be built without requiring additional components.

**Comment by christianvogt** [comment]
@Sam wouldn't this be a good component to expose through the SDK? If it can be placed in any layout, the API is kube related, and the API is simple.

**Comment by glekner** [comment]
@spadgett @christianvogt we need this for our details pages

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts [hunk 1]
```diff
@@ -585,3 +585,7 @@ export type SelfSubjectAccessReviewKind = {
     evaluationError?: string;
   };
 };
+
+export type ResourceEventStreamProps = {
+  obj?: K8sResourceCommon;
```
Comment (christianvogt) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts:590]: I think `resource` is generally used for such a prop. `obj` is too vague.

Comment (christianvogt) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts:590]: Furthermore is `obj` really optional?

Comment (bipuladh) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts]: Adding a wrapper and exposing it as Resource.

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts [hunk 2]
```diff
@@ -144,3 +145,16 @@ export { useFlag } from '../utils/flags';
  */
 export const ResourceYAMLEditor: React.FC<ResourceYAMLEditorProps> = require('@console/internal/components/AsyncResourceYAMLEditor')
   .AsyncResourceYAMLEditor;
+
+/**
+ * A component to show events related to a particular resource.
+ *
+ * @example
+ * tsx```
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:153]:
```suggestion
 * ```tsx
```

**Pull request merged**